src/LinuxTracing/TracerImpl.cpp

```diff
@@ -512,7 +513,11 @@ void TracerImpl::InitSwitchesStatesNamesVisitor() {
   switches_states_names_visitor_ = std::make_unique<SwitchesStatesNamesVisitor>(listener_);
   switches_states_names_visitor_->SetProduceSchedulingSlices(trace_context_switches_);
   if (trace_thread_state_) {
-    switches_states_names_visitor_->SetThreadStatePidFilter(target_pid_);
+    std::set<pid_t> pids = {target_pid_};
+    if (introspection_enabled_) {
```

Comment:
A tiny comment along the lines of the first two lines of the commit message could be interesting? Maybe it's super obvious, but once could be surprised reading this part.

Reply:
done.